vars/pullGitRepositoryToSapCloudPlatformAbapEnvironment.groovy

```diff
@@ -0,0 +1,185 @@
+     * Specifies the host address of the SAP Cloud Platform ABAP Environment system
+     */
+    'host',
+    /**
+     * Specifies the name of the Repository (Software Component) on the SAP Cloud Platform ABAP Environment system
+     */
+    'repositoryName'
+]
+@Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS.plus([
+    /**
+     * Specifies the communication user of the communication scenario SAP_COM_0510
+     */
+    'username',
+    /**
+     * Specifies the password of the communication user
+     */
+    'password'
+])
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+/**
+ * Pulls a Repository (Software Component) to a SAP Cloud Platform ABAP Environment system.
+ *
+ * !!! note "Git Repository and Software Component"
+ *       In SAP Cloud Platform ABAP Environment Git repositories are wrapped in Software Components (which are managed in the App "Manage Software Components")
+ *       Currently, those two names are used synonymous.
+ */
+@GenerateDocumentation
+void call(Map parameters = [:]) {
+
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters, failOnError: true) {
+
+        def script = checkScript(this, parameters) ?: this
+
+        ConfigurationHelper configHelper = ConfigurationHelper.newInstance(this)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+
+        Map configuration = configHelper.use()
+
+        configHelper
+            .withMandatoryProperty('host', 'Host not provided')
+            .withMandatoryProperty('repositoryName', 'Repository / Software Component not provided')
+            .withMandatoryProperty('username')
+            .withMandatoryProperty('password')
+
+        String usernameColonPassword = configuration.username + ":" + configuration.password
+        String authToken = usernameColonPassword.bytes.encodeBase64().toString()
+        String urlString = configuration.host + '/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull'
+        echo "[${STEP_NAME}] General Parameters: URL = \"${urlString}\", repositoryName = \"${configuration.repositoryName}\""
+
+        String urlPullEntity = triggerPull(configuration, urlString, authToken)
+
+        if (urlPullEntity != null) {
+            String finalStatus = pollPullStatus(urlPullEntity, authToken)
+            if (finalStatus != 'S') {
+                error "[${STEP_NAME}] Pull Failed"
+            }
+        }
+    }
+}
+
+private String triggerPull(Map configuration, String url, String authToken) {
+
+    String entityUri = null
+    String headerFile = "headerAuth.txt"
+
+    def xCsrfTokenScript = """#!/bin/bash
+        curl -I -X GET ${url} \
+        -H 'Authorization: Basic ${authToken}' \
+        -H 'Accept: application/json' \
+        -H 'x-csrf-token: fetch' \
+        -D ${headerFile} \
+        | awk 'BEGIN {FS=": "}/^x-csrf-token/{print \$2}'
```

Comment:
not sure if `awk` is available on all runtime environments. In our jenkinsfile-runner (`ppiper/jenkinsfile-runner`) it is. This is some kind of reference-setup, but a user might use another environment. So far `awk` is not used in the piper-lib (`grep -ir awk vars src`). Is it easy to replace by e.g. sed? In this case I would like to ask for replacing it. `sed` is already used e.g. for artifact versioning purposes.

Reply:
Removed awk